test/linalg/dense.jl

```diff
@@ -0,0 +1,4 @@
+using Base.Test
+
+# Check that non-floats are correctly promoted
+@test_approx_eq [1 0 0; 0 1 0]\[1,1] [1;1;0]
```

Comment:
couldn't adding the new file wait until there's more than one line to add to it?